AsyncDisplayKit/ASTableView.mm

```diff
@@ -831,6 +831,25 @@ - (void)rangeController:(ASRangeController *)rangeController didDeleteNodes:(NSA
   }
 }
 
+- (void)rangeController:(ASRangeController *)rangeController didReloadNodes:(NSArray *)nodes atIndexPaths:(NSArray *)indexPaths withAnimationOptions:(ASDataControllerAnimationOptions)animationOptions
+{
+  ASDisplayNodeAssertMainThread();
+  LOG(@"UITableView reloadRows:%ld rows", indexPaths.count);
+
+  if (!self.asyncDataSource) {
+    return; // if the asyncDataSource has become invalid while we are processing, ignore this request to avoid crashes
+  }
+
+  BOOL preventAnimation = animationOptions == UITableViewRowAnimationNone;
+  ASPerformBlockWithoutAnimation(preventAnimation, ^{
+    [super reloadRowsAtIndexPaths:indexPaths withRowAnimation:(UITableViewRowAnimation)animationOptions];
+  });
+
+  if (_automaticallyAdjustsContentOffset) {
```

Comment:
@nguyenhuy @lkzhao I think we should honor it wherever it can be reasonably accommodated, especially inserting and deleting elements, as well as moving or reloading individual items and sections as that may cause them to calculate a different height. I think the main challenge with doing it on a full table reload at the moment is that we break it into a delete and insert, At least historically, and because of the asynchronous delay of the insert it was not always feasible to guarantee.